### PR TITLE
Add guest mode options screen

### PR DESCRIPTION
**Pull request by daehn**
## What's new in this PR?

* Add guest mode options screen.

### Dependencies

- [x] New release of `wire-ios-sync-engine` after the branch `feature/conversation-links` has been merged.
- [x] New release of `wire-ios-sync-data-model` after the branch `feature/conversation-links` has been merged.

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -10,7 +10,7 @@ github "wireapp/PINCache" "2.3-swift3.1"
 github "wireapp/PureLayout" "v3.0.0"
 github "wireapp/TTTAttributedLabel" "2.0.0-swift3.0.2"
 github "wireapp/ZipArchive" "v2.0.8"
-github "wireapp/avs-ios-binaries" "4.1.131"
+github "wireapp/avs-ios-binaries" "4.0.128"
```
Comment (mikeger) [inline Cartfile.resolved]: Seems not related to the PR topic, artifact of merge / `carthage update`?

Comment (daehn) [reply Cartfile.resolved]: Yeah probably, will revert.

**Comment by mikeger** [comment]
@daehn can this be merged now?

**Comment by daehn** [comment]
@mikeger I'll update the dependencies.

File: Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m [hunk 2]
```diff
@@ -442,10 +442,13 @@ - (UIViewController *)participantsController
...
     switch (self.conversation.conversationType) {
         case ZMConversationTypeGroup: {
-            ParticipantsViewController *participantsViewController = [[ParticipantsViewController alloc] initWithConversation:self.conversation];
-            participantsViewController.delegate = self;
-            participantsViewController.zClientViewController = [ZClientViewController sharedZClientViewController];
-            viewController = participantsViewController;
+            // TODO jacob: do it properly
+            GroupDetailsViewController *groupDetailsViewController = [[GroupDetailsViewController alloc] initWithConversation:self.conversation];
+            viewController = groupDetailsViewController;
+//            ParticipantsViewController *participantsViewController = [[ParticipantsViewController alloc] initWithConversation:self.conversation];
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m]: @typfel should be removed?

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+        }
+        if conversation.includesServiceUser {
+            let servicesSection = ServicesSectionController(conversation: conversation, delegate: self)
+            sections.append(servicesSection)
+        }
+        return sections
+    }
+    
+    func conversationDidChange(_ changeInfo: ConversationChangeInfo) {
+        // TODO: Check if `teamOnly` changed.
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift]: Should be possible with the current data model: `changeInfo.allowGuestsChanged`.

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return participants.count
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let user = participants[indexPath.row]
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: GroupDetailsParticipantCell.zm_reuseIdentifier, for: indexPath) as! GroupDetailsParticipantCell
+        
+        cell.configure(with: user)
+        cell.separator.isHidden = participants.count - 1 == indexPath.row
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift]: Would be good to add the brackets to be sure about the operations priority.

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return serviceUsers.count
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let user = serviceUsers[indexPath.row]
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: GroupDetailsParticipantCell.zm_reuseIdentifier, for: indexPath) as! GroupDetailsParticipantCell
+        
+        cell.configure(with: user)
+        cell.separator.isHidden = serviceUsers.count - 1 == indexPath.row
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift]: Same as line 308.

File: Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m (hunk 2 above)
Comment (typfel) [reply Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m]: Yes, I'll clean this up hence the `TODO`

File: Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme [hunk 6]
```diff
@@ -123,7 +123,7 @@
          </CommandLineArgument>
          <CommandLineArgument
             argument = "-ZMBackendEnvironmentType staging"
-            isEnabled = "NO">
+            isEnabled = "YES">
```
Comment (mikeger) [inline Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme]: Please revert.

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -6075,7 +6136,7 @@
 				"$(SRCROOT)/Carthage/Build/iOS/WireProtos.framework",
 				"$(SRCROOT)/Carthage/Build/iOS/ProtocolBuffers.framework",
 				"$(SRCROOT)/Carthage/Build/iOS/Cartography.framework",
-				"$(SRCROOT)/Carthage/Build/iOS/WireSyncEngine.framework",
+				"$(BUILT_PRODUCTS_DIR)/WireSyncEngine.framework",
```
Comment (mikeger) [inline Wire-iOS.xcodeproj/project.pbxproj]: Please revert.
